test/e2e-tests.sh

```diff
@@ -0,0 +1,317 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script runs the end-to-end tests against eventing-kafka built
+# from source.
+
+# If you already have the *_OVERRIDE environment variables set, call
+# this script with the --run-tests arguments and it will use the cluster
+# and run the tests.
+# Note that local clusters often do not have the resources to run 12 parallel
+# tests (the default) as the tests each tend to create their own namespaces and
+# dispatchers.  For example, a local Docker cluster with 4 CPUs and 8 GB RAM will
+# probably be able to handle 6 at maximum.  Be sure to adequately set the
+# MAX_PARALLEL_TESTS variable before running this script.
+
+# Calling this script without any arguments will create a new cluster in
+# project $PROJECT_ID, start Knative eventing system, install resources
+# in eventing-kafka, run the tests and delete the cluster.
+
+# Variables supported by this script:
+# PROJECT_ID:  the GKR project in which to create the new cluster (unless using "--run-tests")
+# MAX_PARALLEL_TESTS:  The maximum number of go tests to run in parallel (via "-test.parallel", default 12)
+
+TEST_PARALLEL=${MAX_PARALLEL_TESTS:-12}
+
+# Include the test-infra e2e script functions (See knative.dev/test-infra/scripts/README.md)
+source $(dirname $0)/../vendor/knative.dev/test-infra/scripts/e2e-tests.sh
+
+# If gcloud is not available make it a no-op, not an error.
+which gcloud &> /dev/null || gcloud() { echo "[ignore-gcloud $*]" 1>&2; }
+
+# Use GNU Tools on MacOS (Requires the 'grep' and 'gnu-sed' Homebrew formulae)
+if [ "$(uname)" == "Darwin" ]; then
+  sed=gsed
+  grep=ggrep
+fi
+
+# Eventing main config path from HEAD.
+readonly EVENTING_CONFIG="./config/"
+readonly EVENTING_MT_CHANNEL_BROKER_CONFIG="./config/brokers/mt-channel-broker"
+readonly EVENTING_IN_MEMORY_CHANNEL_CONFIG="./config/channels/in-memory-channel"
+
```

Comment:
why is this IMC stuff here ?

Reply:
Copied too many lines from eventing-contrib; removed